src/content/developers/docs/scaling/zk-rollups/index.md

```diff
@@ -1,34 +1,206 @@
 ---
 title: Zero-Knowledge Rollups
-description: Introduction to zero-knowledge rollups
+description: An introduction to zero-knowledge rollups, a scaling solution used by the Ethereum community. 
 lang: en
 sidebar: true
----
+--- 
+
+Zero-knowledge rollups (ZK-rollups) are layer 2 [scaling solutions](/developers/docs/scaling/) that improve throughput on Ethereum Mainnet through off-chain execution. ZK-rollups process transactions independently, but post state data to Mainnet as `calldata`. They also produce cryptographic proofs of validity for transaction batches to guarantee the correctness of off-chain computation. 
 
 ## Prerequisites {#prerequisites}
 
 You should have a good understanding of all the foundational topics and a high-level understanding of [Ethereum scaling](/developers/docs/scaling/). Implementing scaling solutions such as rollups is an advanced topic as the technology is less battle-tested and continues to be researched and developed.
 
 Looking for a more beginner-friendly resource? See our [introduction to layer 2](/layer-2/).
 
-## Zero-knowledge rollups {#zk-rollups}
+## What are zero-knowledge rollups? {#what-are-zk-rollups}
+
+**Zero-knowledge rollups (ZK-rollups)** execute transactions on layer 2 (L2), which they bundle (or "roll-up") into batches. ZK-rollup operators produce [validity proofs](/glossary/#validity-proof) to prove the validity of off-chain transactions before submitting batches on-chain. These proofs can come in the form of ZK-SNARKs (Zero-Knowledge Succinct Non-Interactive Argument of Knowledge) or ZK-STARKs (Zero-Knowledge Scalable Transparent Argument of Knowledge). 
+
+The ZK-rollup smart contract maintains the state of all transactions on the L2 chain, and this state can only be updated with a validity proof. This means that ZK-rollups only need to provide validity proofs to finalize transactions on the layer 1 (L1) chain instead of posting all the transaction data like [optimistic rollups](/developers/docs/scaling/optimistic-rollups/). 
```

Comment:
It's a bit early to talk about the "L2 chain" because the overall architecture of a zk rollup hasn't been explained. Suggest including some explicit description and perhaps diagram showing that the rollup is managed by a smart contract on L1 that interacts with some off-chain infrastructure. It should also be noted that "offchain" just means off L1, not necessarily off of any chain.

Reply:
> There is a lot of good information here but I think it needs restructuring to make sure the information is presented in a logical order. There is a lot of assumed knowledge and concepts that are used in earlier sections before they are defined in later sections. I have tried to point out some specific examples and make some suggestions.

> There is a lot of good information here but I think it needs restructuring to make sure the information is presented in a logical order. There is a lot of assumed knowledge and concepts that are used in earlier sections before they are defined in later sections. I have tried to point out some specific examples and make some suggestions.

I can see where you're coming from, but it'll help to clarify a few points:
- Most of the content in the 'What are ZK-rollups?" section were based on the page's original text. If you can look through it  [here](https://ethereum.org/en/developers/docs/scaling/zk-rollups/#zk-rollups), you'll find many references to "layer 1" and  "layer 2". I merely improved on the original to emphasize some ideas.

- The way I saw it, the intro section "What are ZK-rollups?" would serve as a high-level overview to the concept (the original page had that high-level summary feeling, too). I felt it'd be more useful to have a separate section detailing the low-level operations of ZK-rollups for those interested (instead of overloading readers with descriptions from the beginning). 

- Also, I should point out that the page assumes the reader is already familiar with scaling and L2s (this is recommended in the prerequisites). This fact, and the page's position in the advanced section of the docs, means a high-level overview using terms like "calldata" or "L2 chain" isn't entirely out of place.

Reply:
ok fair enough. Let's get some other opinions, e.g. @minimalsm. I still think it is better to include some definitions early on so that the page stands alone as much as possible.